pkg/diff/diff.go

```diff
@@ -635,6 +633,14 @@ func (sc *Syncer) Solve(ctx context.Context, parallelism int, dry bool, isJSONOu
 				}); err != nil {
 					return nil, fmt.Errorf("failed processing auto fields: %w", err)
 				}
+
+				if oldPlugin, ok := e.OldObj.(*state.Plugin); ok {
+					oldPluginCopy := &state.Plugin{Plugin: *oldPlugin.DeepCopy()}
+					e.OldObj = oldPluginCopy
+					if err := kong.ClearUnmatchingDeprecations(&pluginCopy.Plugin, &oldPluginCopy.Plugin, schema); err != nil {
```

Comment:
maybe add a small comment that says what this does and why? Up to you though, not strictly required.

Reply:
Fixed ✅